crates/sui-json-rpc/src/read_api.rs

```diff
@@ -151,6 +155,78 @@ impl RpcFullNodeReadApiServer for FullNodeApi {
         Ok(self.state.dry_exec_transaction(tx_data, txn_digest).await?)
     }
 
+    async fn get_coin_metadata(&self, coin_type: String) -> RpcResult<SuiCoinMetadata> {
```

Comment:
This looks quite expensive? We are pulling out the package object every time we need to get the coin's metadata.

if this is frequently accessed, maybe we can consider creating a mapping in `AuthorityPerpetualTables`?